src/cpu/arm.rs

```diff
@@ -25,11 +25,62 @@ fn detect_features() -> u32 {
     use libc::c_ulong;
 
     // XXX: The `libc` crate doesn't provide `libc::getauxval` consistently
-    // across all Android/Linux targets, e.g. musl.
+    // across all Android/Linux targets, e.g. uclibc.
+    // If weak linking of extern symbols becomes available, this code could
+    // choose to dynamically fall back to the procfs version below.
+    #[cfg(not(target_env = "uclibc"))]
```

Comment:
If uclibc supports this when dynamically linking, then the conditional logic should include `not(all(target_env = "uclibc", target_feature = "crt-static"))`. Since this is specific to Linux and not Android (why would we support uclibc on Android?), AFAICT, then we'd have `not(all(target_env = "uclibc", target_os = "linux", target_feature = "crt-static"))`, or similar?

Reply:
I meant "weak linking" in the sense of defining a symbol which would be non-null when available via the dynamic linker, or null if it were not available at dylib linking time.  This is a feature which Rust has considered adding from time to time, mainly for situations these.  But Rust doesn't have it (yet) so it's sorta a moot point.  I can certainly remove this comment if it's causing confusion.

Reply:
No, the comment about weak linking isn't confusing. (Also we do have weak linking with `dlsym`, right?)

In your other comment, you said we can't rely on uClibc having getauxval even when dynamically linked, so my comment regarding crt-static is moot, probably. However, the part where we should restrict this to linux (not do it for android) is still relevant.